### ISSUE TITLE
Upload to different site collection not working

### ISSUE THREAD

**Issue by visioinc** [purtuga/SPWidgets#67]
It looks like the "webURL" parameter is being dropped from the upload tool.

Debugging, I have the parameter present, but by the time we get back to line 238:
`var options = $.extend({}, callerFn.defaults, opt)... `
"opt" only contains listName, no webURL... so it queries the current site instead.

function @ line 7511 feeding it (upload.getListUID) only takes an input param of listName... no webURL.

spwidgets 2.5.3 (although comment at top of file says v2.5.1)
jquery 1.11.3
jqueryUI 1.11.4

**Comment by purtuga** [comment]
I don't have a good response yet, but just wanted to let you know that I'll try to look at it tonight (or soon)

Paul
-- sent from mobile

**Comment by visioinc** [comment]
No rush. I patched it in our environment, just wanted to point it out for a future bug fix.

Oddly enough, the version I am using came from http://purtuga.github.io/SPWidgets/ download version 2.5.3, but the notes show 2.5.1 with a revision date of 2015-10-31. Version on github shows 2.5.2 with a date of 2015-10-17. I ::think:: I'm using 2.5.3... but not 100% sure. Looking at the github master, it looks like the problem is likely to be there as well.

Here's the fix (line#'s from from 2.5.3):

changed if block @ 6961 from
```javascript
        if (opt.listName && opt.listName.indexOf('{') !== 0) {
          opt.listName = Upload.getListUID(opt.listName);
        }
```
to
```javascript
        if (opt.listName && opt.listName.indexOf('{') !== 0) {
          opt.listName = Upload.getListUID(opt.listName, opt.webURL);
        }
```

changed function @ 7511 from
```javascript
    Upload.getListUID = function (listName) {
      if (!listName) {
        return '';
      }
      var id = '';
      getList({
        listName: listName,
        async: false,
        cacheXML: true,
        completefunc: function (xData) {
          id = $(xData.responseXML).find('List').attr('ID');
        }
      });
      return id;
    };
```
to
```javascript
    Upload.getListUID = function (listName, site) {
      if (!listName) {
        return '';
      }
      var id = '';
      getList({
        listName: listName,
        webURL: site,
        async: false,
        cacheXML: true,
        completefunc: function (xData) {
          id = $(xData.responseXML).find('List').attr('ID');
        }
      });
      return id;
    };
```

I didn't test to see if this broke anything else.

Thanks again. Still get a lot of use out of this awesome tool.

**Comment by purtuga** [comment]
Thank you for that.  I'll correct it soon. 
Appreciate you posting back the solution. 
Paul
-- sent from mobile

**Comment by purtuga** [comment]
Ok.. I got some time to today to apply your fix... I have released v2.5.4... can you try? - https://github.com/purtuga/SPWidgets/releases/tag/v2.5.4

**Comment by visioinc** [comment]
Worked perfect. Thanks!
